messaging/src/main/java/org/axonframework/eventhandling/EventTrackerStatus.java

```diff
@@ -49,6 +51,21 @@
      */
     boolean isReplaying();
 
+    /**
+     * Indicates whether this Segment is still merging events.
+     *
+     * @return {@code true} if this segment is merging events, otherwise {@code false}
+     */
+    boolean isMerging();
+
+    /**
+     * Return the estimated relative token position this Segment will have after a merge operation is complete.
+     * In case no estimation can be given or no merge in progress, an {@code OptionalLong.empty()} will be returned.
+     *
+     * @return return the estimated relative position this Segment will reach after a merge operation is complete.
+     */
+    OptionalLong mergeCompletedPosition();
```

Comment:
I am not overly pleased with the name of this method, but haven't landed on something satisfying either...

I can only think in terms of "upper most token" and "lowest token", which would suggest that this method would be renamed to `upperMergePosition`/`upperMergingPosition` and the other would be `lowestMergePosition`/`lowestMergingPosition`.

What do you think?

Reply:
I'm not happy with the name too, but didn't found nothing that fits better.
From a user perspective upperMergePosition doen't provide the same information as merceCompletedPosition, in my opinion.
Let's think about it for a while.

Reply:
We discussed with @abuijze on this: from this comment https://github.com/AxonFramework/AxonFramework/pull/1330/files#r372591669 on another PR I think we can then have a method called `currentPosition` that exposes the `estimation current tracking token position` if available.

In addition to this we will have specific replay/merge method suck has
`mergeCompletedPosition()` : estimated relative token position this Segment will have after a merge operation is complete
`isMerging()` : Indicates whether this Segment is still merging events
`isReplaying()` : Indicates whether this Segment is still replaying previously processed Events
`getResetPosition()` :  the relative position at which a reset was triggered for this Segment

So `currentPosition` will promote and expose TrackinToken.position() method and return
- the estimated relative replay position this Segment represents (in case of Replay)
- the estimated relative merge position this Segment represents (in case of merge)
- OptionalLong.empty() for all the other case

what do you think?

Reply:
That sounds like a great idea! Think my comment has been overruled by this :+1: